sktime/forecasting/base/_base.py

```diff
@@ -23,10 +59,31 @@ class BaseForecaster(BaseEstimator):
 
     def __init__(self):
         self._is_fitted = False
+
+        self._y = None
+        self._X = None
+
+        # forecasting horizon
+        self._fh = None
+        self._cutoff = None  # reference point for relative fh
+
+        # defaults for estimator tags
+        if not hasattr(self, "_tags"):
+            self._tags = dict()
+
+        if hasattr(self, "fhmixinflag"):
+            self._tags["fh_in_fit"] = self.fhmixinflag
+
+        if "fh_in_fit" not in self._tags.keys():
+            self._tags["fh_in_fit"] = "required"
```

Comment:
Could we not just keep the binary tags and make it `fit-requires-fh`? And could that tag not be defined by the mixin classes, so that we don't have to use the `fhmixinflag`? Or am I missing something here?

Reply:
see #950